src/python/build_management/build_manager.py

```diff
@@ -984,6 +993,44 @@ def delete(self):
     raise BuildManagerException('Cannot delete system build.')
 
 
+class AuxiliaryBuild(Build):
```

Comment:
I thought about that initially, but then it looked too hacky to me, as it seemed like we would have to add some branching / special handling in several places (e.g. `_pre_setup`, `build_update` logic, `_setup_application_path`). Thinking long term, we may have other use cases for auxiliary builds (e.g. a coverage-instrumented build, or some new fuzzing engine that needs different builds), so I thought that those use cases would complicate a `RegularBuild` even further, while with `AuxiliaryBuild` we have more flexibility and lesser risk of landing a breaking CL, so I pivoted to a separate build type.

RPC is kinda annoying to add, but it looks like it's done just once and shouldn't require frequent modifications. Maybe I misunderstood this part?

Reply:
I don't think this will look too hacky if we plumb things properly. In this case, the only change required should be to plumb APP_PATH/APP_DIR/APP_REVISION etc by adding them as arguments?

If we have the ability to pass them as argument, we don't need any hacks in `_pre_setup` etc, because we're not overwriting anything that we care about in the previously set up build. Rather than making things more hacky, this is making RegularBuild more general and useful.

This will enable more auxiliary builds in the future also, and removing this hardcoding on the global BUCKET_PATH/APP_PATH/APP_DIR will make build_manager much more flexible in the future. Right now there is a lot of duplicated logic.

Reply:
Sounds good, thanks for clarifications! Will give it a shot.